src/StatusQ/Controls/StatusInput.qml

```diff
@@ -30,22 +30,43 @@ Item {
     property int charLimit: 0
     property string errorMessage: ""
     property list<StatusValidator> validators
+    property bool dirty: false
+    property int validationMode: StatusInput.ValidationMode.OnlyWhenDirty
+    enum ValidationMode {
+        OnlyWhenDirty, // validates input only after it has become dirty
+        Always // validates input even before it has become dirty
+    }
```

Comment:
I wonder if `OnlyWhenDirty` could lead to undesired behaviour but I'm probably missing something..

At the moment, validation happens `onCompleted` which is needed because, if a field is initialized with a value and its valid, the input will still be `valid: false` until `onTextChanged` was emitted.

Using `OnlyWhenDirty` we re-introduce this ill-behaviour, no?

Reply:
Let's think of the case where we want to validate a field with a min length of 1. If the input loads (ie `Component.onCompleted` is fired) with a blank input, validation will run, and the validator will see that the input which has text of `""`, and it'll say "oh, this isn't valid". However, the input should remain with `valid` = false From a UX perspective, we really don't want that input validated yet as the user hasn't had the chance to add their input. However, if the input was part of an edit form, it would contain pre-populated values, and in that case, it would be useful to the user to see a validation error message.

So we are presented with two cases:
1. a new clean form with inputs that haven't been edited in which we don't want validation to run `onCompleted`
2. a form with pre-populated values that we do want validation to run `onCompleted`

This allows consumers of `StatusInput` to dictate when the validation will occur, which is what I have done in https://github.com/status-im/status-desktop/pull/3425/files#diff-374f8aee4fa4150e3039b4cbfccc3114cfdb704c03045ec0b60a00cb0bbf2e3fR34-R36:
```qml
property int validationMode: edit ?
        StatusInput.ValidationMode.Always :
        StatusInput.ValidationMode.OnlyWhenDirty
```

Reply:
> However, the input should remain with valid = false From a UX perspective, we really don't want that input validated yet as the user hasn't had the chance to add their input.

I see, so this is really just to save computation then? Because all inputs are `valid: false` by default. Having them validate with an empty value cause them to stay `valid: false` (which is the desired behaviour). If I understand correctly, all we're doing with this flag is save the computation because we already "know" the field is invalid (because it hasn't been touched by the user yet).

If the above is true, then there's still a possibility that users won't use the "correct" `mode` (e.g. simply not setting `OnlyWhenDirty` would still work "as expected" [`valid: false` remains]).

Should probably be guided via documentation then.

Anyways, thanks for clarifying. I think we can keep this then.

Reply:
It seems that the base input is `valid: true` by default:
https://github.com/status-im/StatusQ/blob/48309d3040dcaa5a6ee955f8a95153b18316edbc/src/StatusQ/Controls/StatusBaseInput.qml#L47

Reply:
Oh dang. You're right. Wonder if we should change this..